src/Container.php

```diff
@@ -283,7 +284,15 @@ protected function getServiceInstance($service)
     {
         // does the definition exist?
         if (! $this->has($service)) {
-            throw Exception::serviceNotFound($service);
+            if ($this->resolver instanceof AutoResolver) {
+                try {
+                    $this->services[$service] = $this->newInstance($service);
+                } catch (Exception $e) {
+                    throw Exception::serviceNotFound($service);
```

Comment:
@glen-84 as you didn't like #119 , I think this PR is what you are looking. Instead of throwing service not found throw the exception of not able to create and yes has can also return true in that case.